pkg/handlers/nutanix/mutation/prismcentralendpoint/tests/generate_patches.go

```diff
@@ -31,20 +33,52 @@ func TestGeneratePatches(
 			Name: "unset variable",
 		},
 		capitest.PatchTestDef{
-			Name: "all fields set",
+			Name: "all required fields set",
+			Vars: []runtimehooksv1.Variable{
+				capitest.VariableWithValue(
+					variableName,
+					v1alpha1.NutanixPrismCentralEndpointSpec{
+						Host:     "prism-central.nutanix.com",
+						Port:     9441,
+						Insecure: true,
+						Credentials: corev1.LocalObjectReference{
+							Name: "credentials",
+						},
+					},
+					variablePath...,
+				),
+			},
+			RequestItem: request.NewNutanixClusterTemplateRequestItem(""),
+			ExpectedPatchMatchers: []capitest.JSONPatchMatcher{
+				{
+					Operation: "replace",
+					Path:      "/spec/template/spec/prismCentral",
+					ValueMatcher: gomega.SatisfyAll(
+						gomega.HaveKeyWithValue(
+							"address",
+							gomega.BeEquivalentTo("prism-central.nutanix.com"),
+						),
+						gomega.HaveKeyWithValue("port", gomega.BeEquivalentTo(9441)),
+						gomega.HaveKeyWithValue("insecure", true),
+						gomega.HaveKey("credentialRef"),
+						gomega.Not(gomega.HaveKey("additionalTrustBundle")),
+					),
+				},
+			},
+		},
+		capitest.PatchTestDef{
+			Name: "additional trust bundle is set",
 			Vars: []runtimehooksv1.Variable{
 				capitest.VariableWithValue(
 					variableName,
 					v1alpha1.NutanixPrismCentralEndpointSpec{
 						Host:     "prism-central.nutanix.com",
 						Port:     9441,
-						Insecure: false,
+						Insecure: true,
```

Comment:
should be insecure false i think?

Reply:
I wanted to assert in the test that even setting to `Insecure: true` will force it to be `insecure: false` in NutanixClusterTemplate.

Reply:
not sure if we should override user behavior ourselves. flexibility is good to give in my opinion.
would you prefer to add another test. 
- one with insecure false and additiontrustbundle set and 
- other with Insecure: true and additiontrustbundle not set and pc uses self signed and it should not work and expect error (assuming no override?)

Reply:
Also should we add mock data to return errors instead of expecting existing pc so that we can simulate all?